src/data/wax_index.rs

```diff
@@ -15,7 +16,7 @@ use std::collections::HashMap;
 pub struct WaxIndex {
     base_dir: PathBuf,
     #[serde(serialize_with = "toml::ser::tables_last")]
-    index: HashMap<String, String>,
+    index: HashMap<String, (String, String)>,
```

Comment:
Perhaps we can use a struct?

I don't know what the first string and second are for (or perhaps just a comment will suffice).